### PR TITLE
Fetch program

### PR DESCRIPTION
**Pull request by MrRDotS**
Refactored the Profile settings page
-------------------------------------------

- Added capability to update user's name and email 
- Edited options in profile settings page to better reflect the current options available.

### CODE REVIEW

File: lib/api/user_repo.dart [hunk 1]
```diff
@@ -23,21 +23,33 @@ class UserRepositoryImpl implements UserRepository {
...
   }
 
-  static User mockUser() {
-    return new User(name: "test", email: "test@flutter.ca");
+  @override
+  Future<User> updateUsersName(String id, String newName, String email) async {
+    final MutationOptions mutationOptions = MutationOptions(documentNode: gql(UserDocument.updateUsersName(id, newName)));
+    final QueryResult queryResult = await utility.client.mutate(mutationOptions);
+    if (queryResult.hasException) {
+      log("Error fetching profile details ${queryResult.exception}");
```
Comment (boiledbuns) [inline lib/api/user_repo.dart]: Should probably make a bit more specific - something like `error updating user name ...`

File: lib/api/user_repo.dart [hunk 2]
```diff
@@ -23,21 +23,33 @@ class UserRepositoryImpl implements UserRepository {
...
-        name: queryResult.data['user']['name'],
-        email: queryResult.data['user']['email']);
+        name: name,
+        email: queryResult.data['updateUser']['user']['email']);
   }
 
-  static User mockUser() {
-    return new User(name: "test", email: "test@flutter.ca");
+  @override
+  Future<User> updateUsersName(String id, String newName, String email) async {
```
Comment (boiledbuns) [inline lib/api/user_repo.dart]: Should email be returned here? I think we can just update the query to return the full user object and just return a user object parsed from `queryResult.data['updateUser']['user']`

File: lib/api/workout_repo.dart [hunk 3]
```diff
@@ -5,7 +5,8 @@ import 'graphql.dart';
 
 abstract class WorkoutRepository {
   Future<Program> retrieveProgram();
-  Future<Workout> retrieveWorkout();
+
+  Future<Workout> mockRetrieveWorkout();
```
Comment (boiledbuns) [inline lib/api/workout_repo.dart]: just curious but why the change in name? I think it would've been ok to leave it since we would have to undo it once we implement the functionality after we implement it

File: lib/presentation/component/profile_settings.dart [hunk 4]
```diff
@@ -1,12 +1,21 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/widgets.dart';
 import 'package:stronk/presentation/discover/discover_page.dart';
+import 'package:stronk/presentation/profile/UpdateUserDetailsPage.dart';
+import 'package:stronk/presentation/profile/profile_bloc.dart';
 
 class ProfileSettings extends StatelessWidget {
+  final List<String> options = [
```
Comment (boiledbuns) [inline lib/presentation/component/profile_settings.dart:7]: niceee

File: lib/presentation/profile/UpdateUserDetailsPage.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  _renderBody(BuildContext context, ProfileBloc profileBloc) {
+    return Container(
+        child: Column (
+          children: <Widget>[
+            FlatButton(
+                child: Text("Update email"),
+                onPressed: () => {
+                  profileBloc.add(new UpdateEmailInfoEvent(name: profileBloc.state.user.name, newEmail: "new" + profileBloc.state.user.email))
```
Comment (boiledbuns) [inline lib/presentation/profile/UpdateUserDetailsPage.dart:35]: just a small thing but it might be easier to update it with `"mock_email@mock.mock"` or something like that so we don't have to user `profileBloc.state.user.email` here which can be confusing in the future

File: lib/presentation/profile/UpdateUserDetailsPage.dart [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                }
+            ),
+            FlatButton(
+              child: Text("Update name"),
+              onPressed: () => {
+                profileBloc.add(new UpdateNameEvent(newName: "new" + profileBloc.state.user.name, email: profileBloc.state.user.email))
```
Comment (boiledbuns) [inline lib/presentation/profile/UpdateUserDetailsPage.dart:41]: same thing as above

File: lib/presentation/workout_actions/workoutDocument.dart [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      workoutExercises {
+        workoutId, exerciseId, workoutReps, workoutWeights, restTime,
+        exercise {
+          id, name, description
+        }
+      }
+    }
+  """;
+
+  static String getQueryProgram(String id) => """
```
Comment (boiledbuns) [inline lib/presentation/workout_actions/workoutDocument.dart]: nit: should just be query program

File: pubspec.yaml [hunk 8]
```diff
@@ -3,7 +3,7 @@ description: A new Flutter project.
 
 # The following defines the version and build number for your application.
 # A version number is three numbers separated by dots, like 1.2.43
-# followed by an optional build number separated by a +.
+# followed by an optional build number separated by ap +.
```
Comment (boiledbuns) [inline pubspec.yaml]: nit: should revert this typo

**Review by boiledbuns — APPROVED** [review]
LGTM - just a few things to update before merging :fire: :fire: :fire: :fire: 
also looks like ci is failing

File: lib/api/user_repo.dart (hunk 2 above)
Comment (MrRDotS) [reply lib/api/user_repo.dart]: Good point
